tests/components/irm_kmi/test_config_flow.py

```diff
@@ -0,0 +1,118 @@
+"""Tests for the IRM KMI config flow."""
+
+from unittest.mock import MagicMock
+
+from homeassistant.components.irm_kmi.const import CONF_LANGUAGE_OVERRIDE, DOMAIN
+from homeassistant.config_entries import SOURCE_USER
+from homeassistant.const import ATTR_LATITUDE, ATTR_LONGITUDE, CONF_LOCATION
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import FlowResultType
+
+from tests.common import MockConfigEntry
+
+
+async def test_full_user_flow(
+    hass: HomeAssistant,
+    mock_setup_entry: MagicMock,
+    mock_get_forecast_in_benelux: MagicMock,
+) -> None:
+    """Test the full user configuration flow."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": SOURCE_USER}
+    )
+
+    assert result.get("type") == FlowResultType.FORM
+    assert result.get("step_id") == "user"
+
+    result2 = await hass.config_entries.flow.async_configure(
+        result["flow_id"],
+        user_input={CONF_LOCATION: {ATTR_LATITUDE: 50.123, ATTR_LONGITUDE: 4.456}},
+    )
+    assert result2.get("type") == FlowResultType.CREATE_ENTRY
+    assert result2.get("title") == "Brussels"
+    assert result2.get("data") == {
+        CONF_LOCATION: {ATTR_LATITUDE: 50.123, ATTR_LONGITUDE: 4.456}
+    }
+
+
+async def test_user_flow_home(
+    hass: HomeAssistant,
+    mock_setup_entry: MagicMock,
+    mock_get_forecast_in_benelux: MagicMock,
+) -> None:
+    """Test the full user configuration flow."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": SOURCE_USER}
+    )
+    hass.config.latitude = 50.123
+    hass.config.longitude = 4.456
+
+    result = await hass.config_entries.flow.async_configure(
+        result["flow_id"],
+        user_input={CONF_LOCATION: {ATTR_LATITUDE: 50.123, ATTR_LONGITUDE: 4.456}},
+    )
+    assert result.get("type") == FlowResultType.CREATE_ENTRY
+    assert result.get("title") == "Brussels"
```

Comment:
What does this test?

Reply:
This checks that the config entry title is set to the city name returned by the weather provider.  This was used for the name of the weather entity (which has been set to `None` after comments). 
The config entry still needs a title so the city name is used.